worker.go

```diff
@@ -590,8 +598,28 @@ func updateIngress(kubeAdapter *kubernetes.Adapter, lb *loadBalancer, problems *
 	if lb.clusterLocal {
 		dnsName = kubernetes.DefaultClusterLocalDomain
 	} else {
-		// only update ingress if the stack exists and is in a complete state.
-		if lb.stack == nil || !lb.stack.IsComplete() {
+		// only update ingress if the CF stack is in a completed state and the ELB
+		// is in the active state.
+		if lb.stack == nil {
+			log.Infof("CF stack is nil, skipping ingress update")
+			return
+		}
+		if !lb.stack.IsComplete() {
+			log.Infof(
+				"CF stack %q is not in a completed state, skipping ingress update",
+				lb.stack.Name,
+			)
+			return
+		}
+		if lb.state == nil || lb.state.StateCode != elbv2Types.LoadBalancerStateEnumActive {
```

Comment:
I think if you do this as a function in adapter, then you don't need to have elbv2Types in worker.go